app/workers/proactive_project_sync_worker.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class ProactiveProjectSyncWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: :small, lock: :until_executed
+
+  PLATFORMS = %w[NPM Maven].freeze
+  DEFAULT_LIMIT = 1000
+
+  def perform(limit = DEFAULT_LIMIT)
+    projects_query
+      .limit(limit)
+      .each(&:async_sync)
```

Comment:
In case the passed in `limit` is large, using `find_each` will prevent instantiating a ton of models:

```suggestion
      .find_each(&:async_sync)
```

Reply:
Using `find_each` or the other ActiveRecord::Batches methods would throw out the ordering, and I think it makes sense to attempt the staler rows first?